firebase-database/api/android/firebase-database.api

```diff
@@ -68,41 +69,17 @@ public final class dev/gitlive/firebase/database/FirebaseDatabase {
 public final class dev/gitlive/firebase/database/FirebaseDatabase$Companion {
 }
 
-public final class dev/gitlive/firebase/database/NativeDatabaseReference {
-	public final fun child (Ljava/lang/String;)Ldev/gitlive/firebase/database/NativeDatabaseReference;
-	public fun getAndroid ()Lcom/google/firebase/database/DatabaseReference;
-	public synthetic fun getAndroid ()Lcom/google/firebase/database/Query;
-	public final fun getDatabase ()Ldev/gitlive/firebase/database/FirebaseDatabase;
-	public final fun getKey ()Ljava/lang/String;
-	public final fun onDisconnect ()Ldev/gitlive/firebase/database/NativeOnDisconnect;
-	public final fun push ()Ldev/gitlive/firebase/database/NativeDatabaseReference;
-	public final fun removeValue (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun runTransaction (Lkotlinx/serialization/KSerializer;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public static synthetic fun runTransaction$default (Ldev/gitlive/firebase/database/NativeDatabaseReference;Lkotlinx/serialization/KSerializer;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;ILjava/lang/Object;)Ljava/lang/Object;
-	public final fun setValueEncoded (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun updateEncodedChildren (Ldev/gitlive/firebase/internal/EncodedObject;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-}
-
-public final class dev/gitlive/firebase/database/NativeOnDisconnect {
-	public final fun cancel (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun getAndroid ()Lcom/google/firebase/database/OnDisconnect;
-	public final fun getDatabase ()Ldev/gitlive/firebase/database/FirebaseDatabase;
-	public final fun getPersistenceEnabled ()Z
-	public final fun removeValue (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun setValue (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun updateEncodedChildren (Ldev/gitlive/firebase/internal/EncodedObject;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-}
-
 public final class dev/gitlive/firebase/database/OnDisconnect {
 	public final fun cancel (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public final fun getNative ()Ldev/gitlive/firebase/database/NativeOnDisconnect;
 	public final fun removeValue (Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public final fun setEncodedValue (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
```

Comment:
why have these been added?

Reply:
Because we have a bunch of inline methods that need access to the Native classes. Though those had been set as internal and annotated with @PublishedApi, this results in them still poping up in the API specification.

So it's essentially choosing between having the Native... classes in the api or these methods. I want for these because it makes the API "smaller"

Reply:
To clarify: these methods are also marked as`@PublishedApi internal fun` and the access the native classes inside their code. As a result the Native classes no longer need to be Published, but the method does because public inline methods can only call publish/Published code